sklearnex/neighbors/knn_classification.py

```diff
@@ -15,4 +15,164 @@
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = _check_weights(weights)
+else:
+    from sklearn.neighbors.base import SupervisedIntegerMixin as \
+        BaseSupervisedIntegerMixin
+
+    class KNeighborsClassifier_(onedal_KNeighborsClassifier, onedal_KNeighborsMixin,
+                                BaseSupervisedIntegerMixin, BaseClassifierMixin):
+        @_deprecate_positional_args
+        def __init__(self, n_neighbors=5, *,
+                     weights='uniform', algorithm='auto', leaf_size=30,
+                     p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = _check_weights(weights)
+
+class KNeighborsClassifier(KNeighborsClassifier_):
+    @_deprecate_positional_args
+    def __init__(self, n_neighbors=5, *,
+                 weights='uniform', algorithm='auto', leaf_size=30,
+                 p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                 **kwargs):
+        super().__init__(
+            n_neighbors=n_neighbors,
+            weights=weights,
+            algorithm=algorithm,
+            leaf_size=leaf_size, metric=metric, p=p,
+            metric_params=metric_params,
+            n_jobs=n_jobs, **kwargs)
+
+    def fit(self, X, y):
+        X_incorrect_type = isinstance(
+            X, (KDTree, BallTree, onedal_KNeighborsClassifier, BaseNeighborsBase))
+
+        if not X_incorrect_type and self.weights in ['uniform', 'distance'] \
+            and self.algorithm in ['brute', 'kd_tree', 'auto', 'ball_tree'] \
+            and self.metric in ['minkowski', 'euclidean', 'chebyshev', 'cosine']:
+            try:
+                logging.info(
+                    "sklearn.neighbors.KNeighborsClassifier."
+                    "fit: " + get_patch_message("onedal"))
+                result = onedal_KNeighborsClassifier._fit(self, X, y)
+            except RuntimeError:
+                logging.info(
+                    "sklearn.neighbors.KNeighborsClassifier."
+                    "fit: " + get_patch_message("sklearn_after_onedal"))
+                result = BaseNeighborsBase.fit(self, X, y)
+        else:
+            logging.info(
+                "sklearn.neighbors.KNeighborsClassifier."
+                "fit: " + get_patch_message("sklearn"))
+            result = BaseNeighborsBase.fit(self, X, y)
+        return result
+
+    def predict(self, X):
+        onedal_model = getattr(self, '_onedal_model', None)
+        if onedal_model is not None and not sp.issparse(X) \
+            and self.weights in ['uniform', 'distance'] \

```

Comment:
I think onedal_model and check X should be enough

Reply:
fixes